source/extensions/omni.isaac.lab/omni/isaac/lab/utils/noise/noise_model.py

```diff
@@ -18,7 +18,23 @@
 
 
 def constant_noise(data: torch.Tensor, cfg: noise_cfg.ConstantNoiseCfg) -> torch.Tensor:
-    """Constant noise."""
+    """Applies a constant noise bias to a given data set.
+
+    Args:
+        data: The unmodified data set to apply noise to.
+        cfg: The configuration parameters for constant noise.
+
+    Returns:
+        The data modified by the noise parameters provided.
+    """
+
+    # fix tensor device for bias on first call and update config parameters
+    if isinstance(cfg.bias, torch.Tensor):
+        if cfg.bias.device is not data.device:
+            cfg.bias = cfg.bias.to(
+                data.device,
+            )
```

Comment:
This should be zero-overhead if the device is the same I think

```suggestion
    if isinstance(cfg.bias, torch.Tensor):
       cfg.bias =  cfg.bias.to(device=data.device)
```